### ISSUE TITLE
support receive Promise plugin 

### ISSUE THREAD

**Issue by shulandmimi** [farm-fe/farm#1224]
### What problem does this feature solve?

```ts
// farm.config.ts
const loadAsyncJsPlugin = async () => ({ ... });
const loadAsyncArrJsPlugin = async () => [{ ... }, { ... }]
const loadAsyncRustPlugin async () => '@farmfe/rust-plugin';
const loadAsyncArrRustPlugin async () => [['@farmfe/rust-plugin', options], ...];

export default {
	plugins: [loadAsyncJsPlugin(), loadAsyncArrJsPlugin(), loadAsyncRustPlugin(), loadAsyncArrRustPlugin()]
}
```

### What does the proposed API look like?

modify

- [packages/core/src/plugin/index.ts](https://github.com/farm-fe/farm/blob/71b6bab71741b9289b6f5b6c2b330b89c41eaeea/packages/core/src/plugin/index.ts#L12-L57)
- [packages/core/src/plugin/index.ts](https://github.com/farm-fe/farm/blob/71b6bab71741b9289b6f5b6c2b330b89c41eaeea/packages/core/src/plugin/index.ts#L60-L74)

combine these two function

**Comment by DeepeshKalura** [comment]
@shulandmimi  I raised the PR tyres to resolve your issue could please look at it and let me know any improvements needed 

https://github.com/farm-fe/farm/pull/1255

**Comment by shulandmimi** [comment]
@DeepeshKalura Thank you very much for your PR, but I see it has not been updated recently, do you have the energy to achieve the rest of it? I can assign this issue to you if you like.

**Comment by DeepeshKalura** [comment]
@shulandmimi Yeah I would love to work in this issue. The reason I not been updating recently because I was getting a bug while working on this issue `Error: Cannot find module '019171a2'` while running with `pnpm bootstrap`.  Debugging this issue cause lot's of time.

I got a suggestion to re-clone the project and restart the work from discord farm community. Working on that part.

**Comment by ErKeLost** [comment]
> @shulandmimi Yeah I would love to work in this issue. The reason I not been updating recently because I was getting a bug while working on this issue `Error: Cannot find module '019171a2'` while running with `pnpm bootstrap`. Debugging this issue cause lot's of time.是的，我很想在这个问题上工作。我最近没有更新的原因是因为我在运行 `pnpm bootstrap` 时遇到一个 `Error: Cannot find module '019171a2'` 错误。调试此问题会导致大量时间。
> 
> I got a suggestion to re-clone the project and restart the work from discord farm community. Working on that part.我得到了一个建议，可以重新克隆该项目并从 discord 农场社区重新开始工作。在那部分工作。

Has the problem been solved? I didn't reproduce the mistakes you found locally.

**Comment by ErKeLost** [comment]
> > @shulandmimi Yeah I would love to work in this issue. The reason I not been updating recently because I was getting a bug while working on this issue `Error: Cannot find module '019171a2'` while running with `pnpm bootstrap`. Debugging this issue cause lot's of time.
> > I got a suggestion to re-clone the project and restart the work from discord farm community. Working on that part.我得
> 
> Has the problem been solved? I didn't reproduce the mistakes you found locally.

**Comment by ErKeLost** [comment]
Has the problem been solved? I didn't reproduce the mistakes you found locally.

**Comment by DeepeshKalura** [comment]
@ErKeLost Even I can't reproduce later. I just delete the entire code and re-clone it. So, I think problem has been solved for good
